### PR TITLE
Jetty 12.1.x tck error message

### PR DESCRIPTION
**Pull request by janbartel**
This PR fixes the output of ErrorHandler in accordance with TCK expectations. See https://github.com/jetty/jetty.project/issues/11931#issuecomment-2179847257.

There is also a branch for entirely refactoring the servlet api `ErrorHandler` (https://github.com/jetty/jetty.project/tree/jetty-12.1.x-refactor-servlet-error-handler) however that is orthogonal to this fix.

### CODE REVIEW

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java [hunk 1]
```diff
@@ -73,6 +73,14 @@ public boolean errorPageForMethod(String method)
         };
     }
 
+    private static Object getRequestErrorAttribute(HttpServletRequest request, String servletApiName, String jettyName)
```
Comment (gregw) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java]: Why is this method necessary?   
I thought the policy was that we would always set the jetty(core) attribute and then any request wrapper used for an error dispatch to a filter/servlet would intercept the servlet attributes and lookup the jetty ones

So our code should then only ever need to lookup jetty(core) attributes?  Unless there is a case where an application sets them and we are expected to act on them?

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java [hunk 2]
```diff
@@ -73,6 +73,14 @@ public boolean errorPageForMethod(String method)
...
+    {
+        Object o = request.getAttribute(servletApiName);
+        if (o == null)
+            o = request.getAttribute(jettyName);
```
Comment (gregw) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java]:
```suggestion
            o = request.getAttribute(coreAttributeName);
```

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java [hunk 3]
```diff
@@ -73,6 +73,14 @@ public boolean errorPageForMethod(String method)
         };
     }
 
+    private static Object getRequestErrorAttribute(HttpServletRequest request, String servletApiName, String jettyName)
+    {
+        Object o = request.getAttribute(servletApiName);
```
Comment (gregw) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java]:
```suggestion
        Object o = request.getAttribute(servletAttributeName);
```

**Review by gregw — CHANGES_REQUESTED** [review]
If this is a minimal fix just to pass the TCK, then it needs a TODO saying so

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java (hunk 1 above)
Comment (janbartel) [reply jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ErrorHandler.java]: That is part of the refactoring effort, which is a much wider scope than just satisfying the tck.

**Review by gregw — APPROVED** [review]

**Pull request merged**